src/int/leading_zeros.rs

```diff
@@ -16,46 +18,47 @@ pub(crate) fn usize_leading_zeros_default(x: usize) -> usize {
     // because it simplifies the final bisection step.
     let mut x = x;
     // the number of potential leading zeros
-    let mut z = usize::MAX.count_ones() as usize;
+    let mut z = T::BITS as usize;
     // a temporary
-    let mut t: usize;
-    #[cfg(target_pointer_width = "64")]
-    {
+    let mut t: T;
+
+    assert!(T::BITS <= 64);
```

Comment:
```suggestion
    const { assert!(T::BITS <= 64) };
```

Might as well turn this into a comptime error (same for the other `assert!` places

Reply:
Indeed, that was my intention. Just somehow failed to do that...